src/psyclone/kernel_tools.py

```diff
@@ -0,0 +1,225 @@
+                        "(implies '--alg-gen'")
+    parser.add_argument('-ostub', help="filename of created kernel stub code "
+                        "(implies '--stub-gen')")
+    parser.add_argument('-api',
+                        help='choose a particular API from {0}, '
+                             'default \'{1}\'.'
+                        .format(str(Config.get().supported_apis),
+                                Config.get().default_api))
+    parser.add_argument('filename', help='file containing Kernel metadata')
+
+    # Make the default an empty list so that we can check whether the
+    # user has supplied a value(s) later
+    parser.add_argument(
+        '-I', '--include', default=[], action="append",
+        help='path to Fortran INCLUDE or module files')
+    parser.add_argument(
+        '-l', '--limit', dest='limit', default='off',
+        choices=['off', 'all', 'output'],
+        help='limit the Fortran line length to 132 characters (default '
+        '\'%(default)s\'). Use \'all\' to apply limit to both input and '
+        'output Fortran. Use \'output\' to apply line-length limit to output '
+        'Fortran only.')
+
+    parser.add_argument("--config", help="Config file with "
+                        "PSyclone specific options.")
+    parser.add_argument(
+        '-v', '--version', dest='version', action="store_true",
+        help='Display version information ({0})'.format(__VERSION__))
+
+    args = parser.parse_args()
+
+    if args.version:
+        print(f"psyclone-kern version: {__VERSION__}", file=sys.stdout)
+
+    # If no config file name is specified, args.config is none
+    # and config will load the default config file.
+    Config.get().load(args.config)
+
+    # Check API, if none is specified, take the setting from the config file
+    if args.api is None:
+        # No command line option, use the one specified in Config - which
+        # is either based on a parameter in the config file, or otherwise
+        # the default:
+        api = Config.get().api
+    elif args.api not in Config.get().supported_apis:
+        print("Unsupported API '{0}' specified. Supported APIs are "
+              "{1}.".format(args.api, Config.get().supported_apis),
+              file=sys.stderr)
+        sys.exit(1)
+    else:
+        # There is a valid API specified on the command line. Set it
+        # as API in the config object as well.
+        api = args.api
+        Config.get().api = api
+
+    # The Configuration manager checks that the supplied path(s) is/are
+    # valid so protect with a try
+    try:
+        if args.include:
+            Config.get().include_paths = args.include
+        else:
+            # Default is to instruct fparser2 to look in the directory
+            # containing the file being parsed
+            Config.get().include_paths = ["./"]
+    except ConfigurationError as err:
+        print(str(err), file=sys.stderr)
+        sys.exit(1)
+
+    alg = None
```

Comment:
Again, I've done it this way to allow for both options to be used simultaneously. But maybe KISS would imply that I shouldn't have done that. (It will get even more complicated if we come up with another piece of functionality.)

Reply:
What do you think @rupertford, should I re-write so that we only support doing one thing at a time?

Reply:
As mentioned earlier, yes I think so. I can't see why we would want to do both (but I may regret saying that!).

Reply:
It's a lot simpler now.